src/shogun/mathematics/Statistics.cpp

```diff
@@ -814,6 +814,171 @@ SGMatrix<float64_t> CStatistics::sample_from_gaussian(SGVector<float64_t> mean,
 	return S;
 }
 
+CStatistics::SigmoidParamters CStatistics::fit_sigmoid(SGVector<float64_t> scores, SGVector<float64_t> labels)
+{
+	SG_SDEBUG("entering CStatistics::fit_sigmoid()\n")
+
+	REQUIRE(scores.vector, "CStatistics::fit_sigmoid() requires "
+			"scores vector!\n");
+
+	/* count prior0 and prior1 if needed */
+	int32_t prior0=0;
+	int32_t prior1=0;
+	SG_SDEBUG("counting number of positive and negative labels\n")
+	{
+		for (index_t i=0; i<scores.vlen; ++i)
+		{
+			if (labels[i]>0)
+				prior1++;
+			else
+				prior0++;
+		}
+	}
+	SG_SDEBUG("%d pos; %d neg\n", prior1, prior0)
+
+	/* parameter setting */
+	/* maximum number of iterations */
+	index_t maxiter=100;
+
+	/* minimum step taken in line search */
+	float64_t minstep=1E-10;
+
+	/* for numerically strict pd of hessian */
+	float64_t sigma=1E-12;
+	float64_t eps=1E-5;
+
+	/* construct target support */
+	float64_t hiTarget=(prior1+1.0)/(prior1+2.0);
+	float64_t loTarget=1/(prior0+2.0);
+	index_t length=prior1+prior0;
+
+	SGVector<float64_t> t(length);
+	for (index_t i=0; i<length; ++i)
```

Comment:
same as above... too vanilla implementation

Reply:
yes, can't this be done a bit more modern?

Reply:
you can either use `std::transform` or at least use the fact that SGVector has iterator so it can actually do a range-loop

Reply:
Will do, thanks for the suggestion

Reply:
Will do, thanks for the suggestion

Reply:
@vigsterkr I opted to use `std::transform`